### PR TITLE
Updated smartgithg to version `6.5.7`.

### PR DESCRIPTION
**Pull request by jraygauthier**

### CODE REVIEW

File: pkgs/applications/version-management/smartgithg/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@ stdenv.mkDerivation rec {
   src = fetchurl {
     url = "http://www.syntevo.com/download/smartgit/" +
           "smartgit-generic-${the_version}.tar.gz";
```
Comment (vbgl) [inline pkgs/applications/version-management/smartgithg/default.nix]: This url leads to 404: download → download**s**.

**Comment by jraygauthier** [comment]
> This url leads to 404: download → downloads.

That's bad, I get the same this morning when using the direct link with
both new and former url.

The original reason for the update was that it was complaining
of different sha. Now I understand what occurred for both
the original package creator and myself: as we've been
passing through the site's <http://www.syntevo.com/smartgit/download?file=smartgit/smartgit-generic-6_5_7.tar.gz> query link to access the download it
succeed in the delay we tested. Now that the download delay expired,
it doesn't work anymore.

Do anyone know how to proceed for such problem?

**Comment by jraygauthier** [comment]
Should we do something like with virtualbox guest additions:
<https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/virtualization/virtualbox/default.nix#L39>

which fail the build with a message:

~~~
 message = ''
In order to use the extension pack, you need to comply with the VirtualBox Personal Use
and Evaluation License (PUEL) by downloading the related binaries from:
https://www.virtualbox.org/wiki/Downloads
Once you have downloaded the file, please use the following command and re-run the
installation:
nix-prefetch-url file://${name}
'';
~~~

requiring manual intervention (url prefetch) by the user.

**Comment by jraygauthier** [comment]
There is a bunch of packages using the same pattern when I `git grep requireFile`.

**Comment by jraygauthier** [comment]
Seem like I've been looking too far :). Had a discussion with one of the software support rep and he pointed out my error: I forgot to look at difference in the url prefix when updating the version. Now
it uses `downloads` instead of `download`. Not sure if it ever worked. The full url should have been:

<http://www.syntevo.com/downloads/smartgit/smartgit-generic-6_5_7.tar.gz>

I should update the pull request.

**Comment by jraygauthier** [comment]
Deleted the pre-fetched store path:

~~~
nix-store --delete /nix/store/4zxnlfzjs55ylzm6fchfzwi0h8sim9l5-smartgit-generic-6_5_7.tar.gz
~~~

and reinstalled the package. It now works perfectly.

**Comment by geerds** [comment]
Thanks for your commitment!

**Pull request merged**